src/xrpld/app/tx/detail/Batch.cpp

```diff
@@ -0,0 +1,403 @@
+                << "txID: " << hash;
+            return temINVALID_FLAG;
+        }
+
+        if (stx.isFieldPresent(sfTxnSignature))
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn cannot include TxnSignature. "
+                                << "txID: " << hash;
+            return temBAD_SIGNATURE;
+        }
+
+        if (stx.isFieldPresent(sfSigners))
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn cannot include Signers. "
+                                << "txID: " << hash;
+            return temBAD_SIGNER;
+        }
+
+        if (!stx.getSigningPubKey().empty())
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn SigningPubKey must be empty. "
+                                << "txID: " << hash;
+            return temBAD_REGKEY;
+        }
+
+        auto const innerAccount = stx.getAccountID(sfAccount);
+        if (auto const preflightResult = ripple::preflight(
+                ctx.app, ctx.rules, parentBatchId, stx, tapBATCH, ctx.j);
+            preflightResult.ter != tesSUCCESS)
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn preflight failed: "
+                                << transHuman(preflightResult.ter) << " "
+                                << "txID: " << hash;
+            return temINVALID_INNER_BATCH;
+        }
+
+        // Check that the fee is zero
+        if (auto const fee = stx.getFieldAmount(sfFee);
+            !fee.native() || fee.xrp() != beast::zero)
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn must have a fee of 0. "
+                                << "txID: " << hash;
+            return temBAD_FEE;
+        }
+
+        // Check that Sequence and TicketSequence are not both present
+        if (stx.isFieldPresent(sfTicketSequence) ==
+            (stx.getFieldU32(sfSequence) != 0))
+        {
+            JLOG(ctx.j.trace())
+                << "BatchTrace[" << parentBatchId << "]: "
+                << "inner txn must have exactly one of Sequence and "
+                   "TicketSequence. "
+                << "txID: " << hash;
+            return temSEQ_AND_TICKET;
+        }
+
+        // Verify that either Sequence or TicketSequence is present
+        if (!stx.isFieldPresent(sfTicketSequence) &&
+            stx.getFieldU32(sfSequence) == 0)
+        {
+            JLOG(ctx.j.trace()) << "BatchTrace[" << parentBatchId << "]: "
+                                << "inner txn must have either Sequence or "
+                                   "TicketSequence. "
+                                << "txID: " << hash;
+            return temSEQ_AND_TICKET;
```

Comment:
codecov is reporting that this line isn't covered by tests, I'm seeing tests for these errors so not worried about it